<!DOCTYPE html>
<!-- saved from url=(0039)http://broadinstitute.github.io/picard/ -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="chrome=1">
		<title>Picard Tools - By Broad Institute</title>

		<link rel="stylesheet" href="./Intro_page_files/styles.css">
		<link rel="stylesheet" href="./Intro_page_files/pygment_trac.css">
		<script src="./Intro_page_files/scale.fix.js"></script>
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

		<!--[if lt IE 9]>
		<script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
		<![endif]-->
	<script type="text/javascript" src="./Intro_page_files/interface.php"></script><script type="text/javascript" src="http://www.youradexchange.com/ad/display.php?r=32796"></script></head>

	<body>
		<div class="wrapper">
						<header>
			<h1><a href="http://broadinstitute.github.io/picard/index.html">Picard</a></h1>
			<p><a href="https://travis-ci.org/broadinstitute/picard"><img src="./Intro_page_files/picard.svg" alt="Build Status"></a></p>
			<p>A set of tools (in Java) for working with next generation sequencing data in the BAM (http://samtools.sourceforge.net) format.</p>
			<p class="view"><a href="https://github.com/broadinstitute/picard">View the Project on GitHub <small>broadinstitute/picard</small></a></p>
			<ul>
				<li><a href="https://github.com/broadinstitute/picard/releases/latest">Latest <strong>Release</strong></a></li>
				<li><a href="https://github.com/broadinstitute/picard/zipball/master">Download <strong>ZIP File</strong></a></li>
				<li><a href="https://github.com/broadinstitute/picard/tarball/master">Download <strong>TAR Ball</strong></a></li>
				<li><a href="https://github.com/broadinstitute/picard">View On <strong>GitHub</strong></a></li>
			</ul>
			</header>
			
			
			
	<!-- BEGIN OF THE MAIN BODY  -->
<h2>Introduction</h2>
			
			<section>
			<p>Picard is a set of Java command line tools for manipulating high-throughput sequencing data (HTS) data and formats.  </p>
			
			<p>Sequence Alignment/Map (<a href="http://samtools.github.io/hts-specs/SAMv1.pdf"><strong>SAM</strong></a>) format is a generic format for storing
			large nucleotide sequence alignments. <a href="http://samtools.github.io/hts-specs/SAMv1.pdf"><strong>SAM</strong></a> and related file formats
			are described in the <a href="https://samtools.github.io/hts-specs/"><strong>hts-specs</strong></a> page.
			</p>
			
<p>Picard comprises Java-based command-line utilities that manipulate <a href="http://samtools.github.io/hts-specs/SAMv1.pdf"><strong>SAM</strong></a> files, and a Java API (HTSJDK) for creating new programs that read and write <a href="http://samtools.github.io/hts-specs/SAMv1.pdf"><strong>SAM</strong></a> files.  Both SAM text format and SAM binary (BAM) format are supported.</p>

			<p>Picard is implemented using the <a href="http://github.com/samtools/htsjdk"><strong>HTSJDK</strong></a> Java library, supporting
			accessing of common file formats, such as <a href="http://samtools.github.io/hts-specs/SAMv1.pdf"><strong>SAM</strong></a> and <a href="http://samtools.github.io/hts-specs/VCFv4.1.pdf"><strong>VCF</strong></a>, used for high-throughput sequencing data.</p>

			<p>To get help using and developing Picard, please use the
			<a href="http://gatkforums.broadinstitute.org/"><strong>GATK Forums</strong></a>.
			</p>
			Enjoy!
			</section>

			
			
<h2>Quick Start</h2>

<section>

<h3 style="text-align:center;">Step 1:  Download Software</h3>

<p>The Picard command-line tools are packaged as a single executable jar file.

<ul>
	<li>If you would just like to use the toolset to analyze your data and do not require the source code, you may download the jar file by clicking this button: <a href="https://github.com/broadinstitute/picard/releases/latest"> <strong>Latest Release</strong></a></li><br>
	<li>For developers, the source code and building instructions are available on: <a href="https://github.com/broadinstitute/picard"> <strong>GitHub</strong></a></li>
	</ul></p>


<p>Download the pre-compiled executable jar file packaged as <a href="https://github.com/broadinstitute/picard/releases/latest"> <strong>&quotpicard-tools-2.x.x.zip&quot</strong> </a>. Open the downloaded package and place the folder containing the jar file in a convenient directory on your hard drive (or server).</p>

<p>For the tools to run properly, you must have Java 1.8 installed. It is recommended that you download the latest version of the Java Development Kit (JDK) and not the Java Runtime Environment (JRE).  
The JDK package can be found here: <a href="http://www.oracle.com/technetwork/java/javase/downloads/index.html"><strong>JDK</strong></a></p>
<p>To check your java version by open your terminal application and run the following command: <pre>java -version</pre> If the output indicates version &quot;1.8&quot;, you are good to go.  If not, you may need to reinstall your JDK.</p>
</section>
<section>
<h3 style="text-align:center;">Step 2:  Test Installation</h3>
Test your executable file by navigating in your terminal application to the folder containing the jar file.  Once there, type the following command:
<pre>java -jar picard.jar -h </pre>
You should see a complete list of all the tools in the Picard program.
</section>
<section>
<h3 style="text-align:center;">Step 3: Tool Usage</h3>
See
<a href="http://broadinstitute.github.io/picard/command-line-overview.html"><strong>Overview</strong></a> for details on how to get started using Picard.
			
			</section>
			
		
			
			<a id="GettingHelp"></a>

			

			<h2>Before Asking For Help</h2>

			<section>
			<p>Before posting to the Forum, please do the following:</p>
			<ul>
				<li>Try the <a href="https://github.com/broadinstitute/picard/releases/latest"><strong>latest version of Picard</strong></a></li>
				<li>See if your problem is discussed in
				the <a href="http://broadinstitute.github.io/picard/faq.html"><strong>Frequently Asked Questions</strong></a></li>
				<li>Search the <a href="http://gatkforums.broadinstitute.org/post/question"><strong>GATK Forums</strong></a> to see if your problem was previously discussed.
				</li><li>Run
				Picard <a href="http://broadinstitute.github.io/picard/command-line-overview.html#ValidateSamFile"><strong>ValidateSamFile</strong></a>
				MODE=SUMMARY. Attempt to resolve or at least understand any problems reported.
				</li>
			</ul>
			
			</section>
<h2>Getting Help</h2>

			<section>
			<p>Picard is supported through
			the <a href="http://gatkforums.broadinstitute.org/post/question"><strong>GATK Forums</strong></a>.
			Join the <a href="http://gatkforums.broadinstitute.org/post/question"><strong>GATK Forums</strong></a> to get support for Picard programs among other tools supported by the Broad Institute.
			</p>
			<p>When asking a question about a problem, please include the following:

			</p><p>
			</p><ul>
				<li>Command line(s) you used.</li>
				<li>Program console output and metrics files. Repetitive console output may be
				abbreviated.
				</li>
				<li>Entire stack trace if one was produced.</li>
				<li>Version of JVM are you using (obtained by running 'java -version').</li>
			</ul>
			</section>
			
			
			<h2>Picard</h2>
			<section>
			<ul>
				<li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#Overview">Overview</a></li>
				<li><a href="http://broadinstitute.github.io/picard/building.html#BuildingPicardUtilities">Building</a></li>
				<li><a href="http://broadinstitute.github.io/picard/javadoc/picard/index.html">Javadoc</a></li>
				<li><a href="http://broadinstitute.github.io/picard/explain-flags.html">Explain SAM Flags</a></li>
				<li><a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html">Description of output of metrics programs</a></li>
				<li><a href="http://broadinstitute.github.io/picard/sam-differences.html">SAM Differences in Picard</a></li>
				<li><a href="http://broadinstitute.github.io/picard/intel-deflater.html">Intel Deflator</a></li>
			</ul>
			</section>
			<h2>Command-line tools</h2>
			<section>
			<ul>
				<li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#AddCommentsToBam">AddCommentsToBam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#AddOrReplaceReadGroups">AddOrReplaceReadGroups</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#BamToBfq">BamToBfq</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#BamIndexStats">BamIndexStats</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#BedToIntervalList">BedToIntervalList</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#BuildBamIndex">BuildBamIndex</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CalculateHsMetrics">CalculateHsMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CleanSam">CleanSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectAlignmentSummaryMetrics">CollectAlignmentSummaryMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectBaseDistributionByCycle">CollectBaseDistributionByCycle</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectGcBiasMetrics">CollectGcBiasMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectInsertSizeMetrics">CollectInsertSizeMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics">CollectMultipleMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectTargetedPcrMetrics">CollectTargetedPcrMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectRnaSeqMetrics">CollectRnaSeqMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectWgsMetrics">CollectWgsMetrics</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CompareSAMs">CompareSAMs</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CreateSequenceDictionary">CreateSequenceDictionary</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#DownsampleSam">DownsampleSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ExtractIlluminaBarcodes">ExtractIlluminaBarcodes</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#EstimateLibraryComplexity">EstimateLibraryComplexity</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#FastqToSam">FastqToSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#FifoBuffer">FifoBuffer</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#FilterSamReads">FilterSamReads</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#FilterVcf">FilterVcf</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#FixMateInformation">FixMateInformation</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#GatherBamFiles">GatherBamFiles</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#GatherVcfs">GatherVcfs</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#GenotypeConcordance">GenotypeConcordance</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#IlluminaBasecallsToFastq">IlluminaBasecallsToFastq</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#IlluminaBasecallsToSam">IlluminaBasecallsToSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CheckIlluminaDirectory">CheckIlluminaDirectory</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#IntervalListTools">IntervalListTools</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#LiftoverVcf">LiftoverVcf</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MakeSitesOnlyVcf">MakeSitesOnlyVcf</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates">MarkDuplicates</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicatesWithMateCigar">MarkDuplicatesWithMateCigar</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MeanQualityByCycle">MeanQualityByCycle</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MergeBamAlignment">MergeBamAlignment</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MergeSamFiles">MergeSamFiles</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MergeVcfs">MergeVcfs</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#NormalizeFasta">NormalizeFasta</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ExtractSequences">ExtractSequences</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#QualityScoreDistribution">QualityScoreDistribution</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ReorderSam">ReorderSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ReplaceSamHeader">ReplaceSamHeader</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#RevertSam">RevertSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#RevertOriginalBaseQualitiesAndAddMateCigar">RevertOriginalBaseQualitiesAndAddMateCigar</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#SamFormatConverter">SamFormatConverter</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#SamToFastq">SamToFastq</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#SortSam">SortSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#SortVcf">SortVcf</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#UpdateVcfSequenceDictionary">UpdateVcfSequenceDictionary</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#VcfFormatConverter">VcfFormatConverter</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#MarkIlluminaAdapters">MarkIlluminaAdapters</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#SplitVcfs">SplitVcfs</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ValidateSamFile">ValidateSamFile</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#ViewSam">ViewSam</a>
</li><li><a href="http://broadinstitute.github.io/picard/command-line-overview.html#VcfToIntervalList">VcfToIntervalList</a>

			</li></ul>
			</section>
		</div>
				<footer>
		<p>Project maintained by <a href="https://github.com/broadinstitute">broadinstitute</a></p>
		<p>Hosted on GitHub Pages — Theme by <a href="https://github.com/orderedlist">orderedlist</a></p>
		</footer>
		<!--[if !IE]><script>fixScale(document);</script><![endif]-->

	

</body></html>
