---
---
<!doctype html>
<html>
{% include head.html title="Picard Tools - By Broad Institute" %}
<body>
<div class="wrapper">
	{% include header.html %}

	<section>

		<p>A set of Java command line tools for manipulating high-throughput sequencing data (HTS) data and formats.  </p>

		<p>Picard is implemented using the HTSJDK Java library <a href="http://github.com/samtools/htsjdk">HTSJDK</a>, supporting
			accessing of common file formats, such as <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a> and <a href="http://samtools.github.io/hts-specs/VCFv4.1.pdf">VCF</a>, used for high-throughput
			sequencing data.</p>
	</section>

	<!-- BEGIN OF THE MAIN BODY  -->

	<h2>Introduction</h2>

	<section>
		<p>SAM (Sequence Alignment/Map) format is a generic format for storing
			large nucleotide sequence alignments. SAM and related file formats
			are described in the <a href="https://samtools.github.io/hts-specs/">hts-specs</a> page.
		</p>

		<p>Picard comprises Java-based command-line utilities that manipulate SAM files,
			and a Java API (HTSJDK) for creating new programs that read and write SAM files.
			Both SAM text format and SAM binary (BAM) format are supported.</p>

		<p>To get help using and developing Picard, please use the
			<a href="http://gatkforums.broadinstitute.org/">GATK Forums</a>.
		</p>
		<p>Getting started: The Picard command-line tools are packaged as a single executable jar file.  You can download the executable jar file by clicking the button at the top of the page labeled &quotLatest Release&quot. For those who would just like to obtain the executable file to use it to analyze their data, just download the folder labeled picard-tools-2.x.x.zip. Open the downloaded file and place the folder containing the jar file in a convenient directory on your hard drive (or server).

		<p>For the tools to run properly, you must have Java 1.8 installed. It is recommended that you download the latest version of the Java Development Kit (JDK) and not the Java Runtime Environment.  The JDK package can be found here: <pre>http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html</pre>  After installation, you need to verify that java has been installed properly.  You can check your java version by opening your terminal and typing: <pre>java -version</pre> If the output indicates version &quot;1.8&quot;, you are good to go.

		<p>The next step entails navigating to your jar file.  You can use the change directory (cd) command in your terminal to navigate to the directory where you placed the folder containing the jar file. Once there, you can type the following command:
		<pre>java -jar picard.jar -h </pre>
		You should see a complete list of all the tools in the Picard program.

		<p>The format for invoking a specific tool is as follows:
		<pre>java <i>jvm-args</i> -jar picard.jar <i>PicardToolName</i> OPTION1=value1 OPTION2=value2...</pre>
		<p>Most of the commands are designed to run in 2GB of JVM, so the <pre><i>jvm-args</i></pre> of &quotXmx2g&quot is recommended.</p>

		<p>The values of each &quotOPTION&quot or arguments can be files, strings, or an integer.  For example, for an argument such as &quotINPUT=input.bam&quot, where &quotINPUT&quot is the option and &quotinput.bam&quot is the file.  Another type of &quotOPTION&quot can be a string; &quotSTRANDSPECIFICITY=NONE&quot where &quotSTRANDSPECIFICITY&quot can be one of several choices including: {NONE, FIRST_READ_TRANSCRIPTION_STRAND, SECOND_READ_TRANSCRIPTION_STRAND}.  For most string options, a default value is automatically invoked but can be changed by the user if required.  Some arguments call for an integer, such as MINIMUM_LENGTH=&quotInteger&quot where &quotInteger&quot is a numerical value such as 500.</p>

		Please note that most files used by the program as &quotINPUTS&quot and &quotOUTPUTS&quot, require a file path e.g. <pre>INPUT=/my_data/input.bam</pre>  where the path is &quot/my_data/&quot and &quot/input.bam&quot is the file name.

		<p>A complete example for using the CollectAlignmentMetrics tools is as follows:
		<pre>java -jar picard.jar CollectAlignmentMetrics \<br>
                   R=/my_data/reference.fasta \<br>
                   I=/my_data/input.bam \<br>
                   O=/my_data/output.txt</pre>

		<p>Picard is hosted by <a target="_blank" href="http://github.com">Github.com</a>.
			The source code is available at the Picard project page
			<a target="_blank" href="http://github.com/broadinstitute/picard">here</a>.

		<p>You can check out the latest source code with:</p>
		<pre>git clone git@github.com:broadinstitute/picard.git</pre>

		<p>The source code for HTSJDK is available on GitHub <a href="https://github.com/samtools/htsjdk">here</a>.
			You can check out the latest source code with:
		</p>
		<pre>git clone git@github.com:samtools/htsjdk.git</pre>
		<p>Enjoy!</p>
	</section>

	<h2>Quick Start</h2>
	<section>
		<p>See <a href="building.html">Building</a> and <a href="command-line-overview.html">Overview</a> for details.</p>
		<!--TODO figure out how to make multi-line <pre> formatting work without resorting to the <br> tag.-->
		<pre>export $JAVA_HOME=<i>Java_6_or_7_install_path</i><br>ant -lib lib/ant package-commands<br>java <i>jvm-args</i> -jar dist/picard.jar <i>PicardCommandName</i> OPTION1=value1 OPTION2=value2...</pre>
	</section>

	<a id="GettingHelp"></a>

	<h2>Getting Help</h2>

	<section>
		<p>Picard is supported through
			the <a href="http://gatkforums.broadinstitute.org/post/question">GATK Forums</a>.
			Join the GATK Forum to get support for Picard programs among other tools supported by the Broad Institute.
		</p>
	</section>

	<h2>Before Asking For Help</h2>

	<section>
		<p>Before posting to the Forum, please do the following:</p>
		<ul>
			<li>Try the <a href="https://github.com/broadinstitute/picard/releases/latest">latest version of Picard</a></li>
			<li>See if your problem is discussed in
				the <a href="faq.html">Frequently Asked Questions</a></li>
			<li>Search the <a href="http://gatkforums.broadinstitute.org/post/question">GATK Forums</a> to see if your problem was previously discussed.
			<li>Run
				Picard<a href="command-line-overview.html#ValidateSamFile">ValidateSamFile</a>
				MODE=SUMMARY. Attempt to resolve or at least understand any problems reported.
			</li>
		</ul>
		<p>When sending email about a problem, please include the following:

		<p>
		<ul>
			<li>Command line(s) you used.</li>
			<li>Program console output and metrics files. Repetitive console output may be
				abbreviated.
			</li>
			<li>Entire stack trace if one was produced.</li>
			<li>Version of JVM are you using (obtained by running 'java -version').</li>
		</ul>
	</section>

	<h2>General Information</h2>
	<section>
		<ul>
			<li><a href="index.html#GettingHelp">Getting Help</a></li>
			<li><a href="faq.html">Frequently Asked Questions</a></li>
			<li><a href="http://github.com/broadinstitute/picard/releases">Download Page</a></li>
			<li>SAMtools</li>
			<ul>
				<li><a href="http://samtools.sourceforge.net/">SAMtools Home Page</a></li>
				<li><a href="http://sourceforge.net/mail/?group_id=246254">SAMtools Mailing Lists</a></li>
			</ul>
			<li>Github Project Pages:</li>
			<ul>
				<li><a href="https://github.com/broadinstitute/picard">Picard</a></li>
				<li><a href="https://github.com/samtools/htsjdk">HTSJDK</a></li>
				<li><a href="https://github.com/samtools/samtools">SAMtools</a></li>
			</ul>
			<li><a href="https://samtools.github.io/hts-specs">SAM/BAM and Related File Format Specifications</a></li>
		</ul>
	</section>
	<h2>Picard</h2>
	<section>
		<ul>
			<li><a href="command-line-overview.html#Overview">Overview</a></li>
			<li><a href="building.html#BuildingPicardUtilities">Building</a></li>
			<li><a href="javadoc/picard/index.html">Javadoc</a></li>
			<li><a href="explain-flags.html">Explain SAM Flags</a></li>
			<li><a href="picard-metric-definitions.html">Description of output of metrics programs</a></li>
			<li><a href="sam-differences.html">SAM Differences in Picard</a></li>
			<li><a href="intel-deflater.html">Intel Deflator</a></li>
		</ul>
	</section>
	<h2>Command-line tools</h2>
	<section>
		<ul>
			{% include command-line-sidebar.html %}
		</ul>
	</section>
</div>
{% include footer.html %}
</body>
</html>
